onnxruntime/core/optimizer/mobile_transformer.cc

```diff
@@ -0,0 +1,240 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <deque>
+#include "core/graph/graph_utils.h"
+#include "core/optimizer/initializer.h"
+#include "core/optimizer/mobile_transformer.h"
+#include "core/mlas/inc/mlas.h"
+#include "core/graph/node_attr_utils.h"
+#include "core/optimizer/utils.h"
+
+using namespace ONNX_NAMESPACE;
+using namespace ::onnxruntime::common;
+namespace onnxruntime {
+namespace {
+
+namespace selectors {
+const Node* GetLoneConsumerNode(const GraphViewer& graph_viewer, const Node& node) {
+  if (!optimizer_utils::CheckOutputEdges(graph_viewer.GetGraph(), node, 1)) {
+    return nullptr;
+  }
+  return &*node.OutputNodesBegin();
+}
+
+class ConvAddActivation : public NodeSelector {
+ public:
+  ConvAddActivation() = default;
+
+  std::optional<NodesToOptimizeIndices> Select(const GraphViewer& graph_viewer, const Node& node) const override {
+    const std::string_view node_ep = node.GetExecutionProviderType();
+    if (node_ep != kCpuExecutionProvider) {
+      return std::nullopt;
+    }
+    //we can't assign `node` as the pre-conv-node, even it is, because the node in this branch may not satisfy requirements
+    const Node* conv_node = nullptr;
+    const auto* add_node = GetLoneConsumerNode(graph_viewer, node);
+    if (!add_node) {
+      return std::nullopt;
+    }
+    // The following transforms only run when the input edge count has already
+    // been decremented to zero by earlier transforms.
+    // any element-wise op could be fused here.
+    if (graph_utils::IsSupportedOptypeVersionAndDomain(*add_node, "Add", {7, 13, 14}) ||
+        graph_utils::IsSupportedOptypeVersionAndDomain(*add_node, "Sum", {6, 8, 13})) {
+      conv_node = MultiSelectBinary(*add_node, true);
+    } else if (graph_utils::IsSupportedOptypeVersionAndDomain(*add_node, "Mul", {7, 13, 14})) {
+      conv_node = MultiSelectBinary(*add_node, false);
+    }
+    if (!conv_node) {
+      return std::nullopt;
+    }
+    const auto* act_node = GetLoneConsumerNode(graph_viewer, *add_node);
+    //even the next node is not a activation node, it's also fine.
+    if (!act_node) {
+    } else if (graph_utils::IsSupportedOptypeVersionAndDomain(*act_node, "Relu", {6, 13, 14}) ||
+                     graph_utils::IsSupportedOptypeVersionAndDomain(*act_node, "Sigmoid", {6, 13}) ||
+                     graph_utils::IsSupportedOptypeVersionAndDomain(*act_node, "Tanh", {6, 13}) ||
+                     graph_utils::IsSupportedOptypeVersionAndDomain(*act_node, "LeakyRelu", {6})) {
+    } else if (graph_utils::IsSupportedOptypeVersionAndDomain(*act_node, "Clip", {6, 11, 12, 13})) {
```

Comment:
A comment saying the contents of the branch are deliberately empty so act_node remains as is would be helpful.